syntax = "proto3";

package rpc;

option go_package = "rpc";

service Firestore {
  rpc AddDocument (Document) returns (DocumentResponse) {}
  rpc GetDocument (DocumentRequest) returns (DocumentResponse) {}
  rpc UpdateDocument(DocumentUpdate) returns (Empty) {}
  rpc DeleteDocument(DocumentRequest) returns (Empty) {}
}

message DocumentRequest {
  string code = 1;
}

message DocumentResponse {
  string name = 1;
}

message DocumentUpdate {
  string code = 1;
  string name = 2;
}

message Empty {}

message Document {
  string code = 1;
  string name = 2;
}